django/db/backends/oracle/compiler.py

```diff
@@ -0,0 +1,55 @@
+from django.db.models.sql import compiler
+
+
+class SQLCompiler(compiler.SQLCompiler):
+    pass
+
+
+class SQLInsertCompiler(compiler.SQLInsertCompiler, SQLCompiler):
+    pass
+
+
+class SQLDeleteCompiler(compiler.SQLDeleteCompiler, SQLCompiler):
+    pass
+
+
+class SQLUpdateCompiler(compiler.SQLUpdateCompiler, SQLCompiler):
+    pass
+
+
+class SQLAggregateCompiler(compiler.SQLAggregateCompiler, SQLCompiler):
+    pass
+
+
+class SQLCheckCompiler(compiler.SQLCheckCompiler, SQLCompiler):
+    # Oracle doesn't support boolean types yet PL/SQL does, meaning that although we
+    # can't compare boolean expressions to NULL in SELECT or WHERE clauses we can in
+    # a function which we can then call from a WHERE clause.
+    check_sql = """\
+WITH
+  FUNCTION f RETURN NUMBER IS
+    b BOOLEAN;
+  BEGIN
+    b := COALESCE(%s, TRUE);
+    IF b THEN
+        RETURN 1;
+    ELSE
+        RETURN 0;
+    END IF;
+  END f;
+SELECT 1 FROM dual WHERE f = 1
+"""
+
+    def as_sql(self):
+        # Avoid case wrapping.
+        self.connection.vendor = "sqlite"
+        condition, params = self.compile(self.query.where)
+        self.connection.vendor = "oracle"
```

Comment:
This is ugly but I can't think of another way of doing that.

Reply:
Yeah if there was some way that the expressions doing the case wrapping with `as_oracle()` to see that it's the check compiler then it could be skipped.